fundingmanager_test.go

```diff
@@ -0,0 +1,1021 @@
+
+	lnw, err := createTestWallet(tempTestDir, netParams,
+		chainNotifier, wc, signer, bio, estimator)
+	if err != nil {
+		t.Fatalf("unable to create test ln wallet: %v", err)
+	}
+
+	arbiterChan := make(chan *lnwallet.LightningChannel)
+	var chanIDSeed [32]byte
+
+	f, err := newFundingManager(fundingConfig{
+		IDKey:        pubKey,
+		Wallet:       lnw,
+		Notifier:     chainNotifier,
+		FeeEstimator: estimator,
+		SignMessage: func(pubKey *btcec.PublicKey, msg []byte) (*btcec.Signature, error) {
+			return nil, nil
+		},
+		SendAnnouncement: func(msg lnwire.Message) error {
+			sentAnnouncements <- msg
+			return nil
+		},
+		ArbiterChan: arbiterChan,
+		SendToPeer: func(target *btcec.PublicKey, msgs ...lnwire.Message) error {
+			sentMessages <- msgs[0]
+			return nil
+		},
+		FindPeer: func(peerKey *btcec.PublicKey) (*peer, error) {
+			return nil, nil
+		},
+		TempChanIDSeed: chanIDSeed,
+		FindChannel: func(chanID lnwire.ChannelID) (*lnwallet.LightningChannel, error) {
+			// This is not expected to be used in the current tests.
+			// Add an implementation if that changes.
+			t.Fatal("did not expect FindChannel to be called")
+			return nil, nil
+		},
+	})
+	if err != nil {
+		t.Fatalf("failed creating fundingManager: %v", err)
+	}
+
+	return f, nil
+}
+
+func recreateAliceFundingManager(t *testing.T, alice *testNode) {
+	// Stop the old fundingManager before creating a new one.
+	if err := alice.fundingMgr.Stop(); err != nil {
+		t.Fatalf("unable to stop old fundingManager: %v", err)
+	}
+
+	aliceMsgChan := make(chan lnwire.Message)
+	aliceAnnounceChan := make(chan lnwire.Message)
+
+	oldCfg := alice.fundingMgr.cfg
+
+	f, err := newFundingManager(fundingConfig{
+		IDKey:        oldCfg.IDKey,
+		Wallet:       oldCfg.Wallet,
+		Notifier:     oldCfg.Notifier,
+		FeeEstimator: oldCfg.FeeEstimator,
+		SignMessage: func(pubKey *btcec.PublicKey,
+			msg []byte) (*btcec.Signature, error) {
+			return nil, nil
+		},
+		SendAnnouncement: func(msg lnwire.Message) error {
+			aliceAnnounceChan <- msg
+			return nil
+		},
+		ArbiterChan: oldCfg.ArbiterChan,
+		SendToPeer: func(target *btcec.PublicKey,
+			msgs ...lnwire.Message) error {
+			aliceMsgChan <- msgs[0]
+			return nil
+		},
+		FindPeer: func(peerKey *btcec.PublicKey) (*peer, error) {
+			return nil, nil
+		},
+		TempChanIDSeed: oldCfg.TempChanIDSeed,
+		FindChannel:    oldCfg.FindChannel,
+	})
+	if err != nil {
+		t.Fatalf("failed recreating aliceFundingManager: %v", err)
+	}
+
+	alice.fundingMgr = f
+	alice.msgChan = aliceMsgChan
+	alice.announceChan = aliceAnnounceChan
+
+	if err = f.Start(); err != nil {
+		t.Fatalf("failed starting fundingManager: %v", err)
+	}
+}
+
+func setupFundingManagers(t *testing.T) (*testNode, *testNode) {
+	// Must init logger to avoid crash.
+	logDir, err := ioutil.TempDir("", "tempLogDir")
+	if err != nil {
+		t.Fatalf("unable to create temp directory: %v", err)
+	}
+
+	logfile := filepath.Join(logDir, "log")
+	initLogRotator(logfile)
```

Comment:
If you're disabling the individual loggers, then you don't _also_ need to initialize the log rotator itself.